.github/workflows/build-and-publish.yml

```diff
@@ -11,106 +11,53 @@ on:
-          $hasChanges = $projChanged.Length -gt 0
-          # Set singleProject to true if projChanged is not already an array of projects.
-          $singleProject = $projChanged -isnot [array]
+          # Gets package name from the release tag.
+          packageName=${releaseTag%%?[[:digit:]]*}
+          echo "package name: $packageName"
+          echo ::set-output name=package_name::$packageName
 
-          # If there are changed projects, convert to JSON array.
-          if ($hasChanges)
-          {
-            # If only one project changed, convert to single element array.
-            if ($singleProject)
-            {
-              $projChangedJSON = ConvertTo-Json -Compress -AsArray $projChanged
-            }
-            # Else, it will already be an array.
-            else
-            {
-              $projChangedJSON = ConvertTo-Json -Compress $projChanged
-            }
-          }
-          echo "project(s) changed JSON - $projChangedJSON"
-          
-          # Set output.
-          echo "::set-output name=has_changes::$hasChanges"
-          echo "::set-output name=diff::$projChangedJSON"
-
-      # Using Github script action to fail the workflow if no project file has been updated.
-      - name: Verify if any project(s) updated
-        uses: actions/github-script@v6
-        with:
-          script: |
-            if ("${{ steps.check-diff.outputs.has_changes }}" == "True") {
-              console.log("Project(s) have been updated");
-            } else {
-              core.setFailed("No project has been updated");
-            }
-      
-      # Using GitHub upload-artifact action to persist the workspace for the next job.
-      - name: Upload workspace
-        uses: actions/upload-artifact@v3
-        with:
-          name: workspace
-          path: ./
+          # Gets package version from the release tag.
+          packageVersion=${releaseTag#${releaseTag%%[0-9]*}}
+          echo "package version: $packageVersion"
+          echo ::set-output name=package_version::$packageVersion
 
   # build job builds for each of the target frameworks of the solution.
   build:
-    name: Build projects
-    # This job will not run until check_diff is complete and passes.
-    needs: check_diff
-    # Using the latest Windows GitHub-hosted runner.
-    runs-on: windows-latest
-    # The matrix is used to define the frameworks that the project(s) support.
-    # For each value in the matrix a concurrent version of the job is run using that value.
-    # If any version of the job fails, it will cancel the other jobs running.
-    strategy:
-      matrix:
-        target_framework: [ net6.0, netcoreapp3.1, net48 ]
+    name: Build ${{ needs.set_variables.outputs.package_name }}
+    # This job will not run until set_variables is complete and passes.
+    needs: set_variables
+    # Using the latest Ubuntu GitHub-hosted runner.
+    runs-on: ubuntu-latest
```

Comment:
Curious about this for build and I wonder if it will cause an issue once we roll it out every repo. 

According to this its got MSBuild 16.10.1.31701 (from /usr/lib/mono/msbuild/15.0/bin/MSBuild.dll)  https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md

Reply:
I forgot to switch this back to `windows-latest` for the build job. I had tested the build job with both the windows and ubuntu runner and they both worked.  However, I did run into an issue with ubuntu runner in the test workflow.